keiko-sql/src/main/kotlin/com/netflix/spinnaker/q/sql/SqlQueue.kt

```diff
@@ -446,6 +448,17 @@ class SqlQueue(
     }
   }
 
+  override fun ackAndPush(message: Message, delay: TemporalAmount) {
+    val fingerprint = message.hashV2()
+    log.debug("Acking and re-pushing message: $message, " +
+      "fingerprint: $fingerprint to delivery in $delay")
+    withRetry(RetryCategory.WRITE) {
+      ackMessage(fingerprint)
+      push(message, delay)
```

Comment:
Looking at what `ackMessage` and `push` do here, I think it makes sense to instead do this here:

Transactionally delete from unacked and insert on queue; outside of the transaction update `messagesTable. updatedAtField`.